sw/device/silicon_creator/lib/drivers/spi_device.h

```diff
@@ -0,0 +1,67 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_SPI_DEVICE_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_SPI_DEVICE_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/silicon_creator/lib/error.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Bit field definitions for JEDEC device ID.
+ *
+ * Note: spi_device sends bits 7:0 of the device ID first.
+ */
+#define SPI_DEVICE_DEV_ID_ROM_BOOTSTRAP_BIT 0
+#define SPI_DEVICE_DEV_ID_CHIP_REV_FIELD \
+  ((bitfield_field32_t){.mask = 0x7, .index = 1})
```

Comment:
Consider the difference between putting the bootstrap bit at bit0 vs bit3:

At bit0, when running owner firmware, the ID field for different revs of the chip will be 0, 2, 4, 6, ...
At bit3, when running owner firmware, the ID field for different revs will be 0, 1, 2, 3, ...

The difference is basically meaningless to upstream devices which will have tables of recognizable IDs.  However, to a person reading a hexdump, it may be more intuitive to put the bootstrap indicator as bit3 so that chip revs appear sequential.

Reply:
Moved this to bit 3, we can adjust later if needed.